mdbenchmark/mdengines/namd.py

```diff
@@ -0,0 +1,196 @@
+
+    Returns
+    -------
+    float
+        nanoseconds per day
+    """
+    lines = fh.readlines()
+
+    for line in lines:
+        if 'Benchmark time' in line:
+            return 1 / float(line.split()[7])
+
+    return np.nan
+
+
+def parse_ncores(fh):
+    """parse number of cores from a NAMD log file
+
+    Parameters
+    ----------
+    fh : str / filehandle
+        filename or string of log file to read
+
+    Returns
+    -------
+    float
+        number of cores job was run on
+    """
+    lines = fh.readlines()
+
+    for line in lines:
+        if 'Benchmark time' in line:
+            return int(line.split()[3])
+
+    return np.nan
+
+
+def analyze_run(sim):
+    """
+    Analyze Performance data of a NAMD simulation
+    """
+    ns_day = np.nan
+    ncores = np.nan
+
+    # search all output files
+    output_files = glob(os.path.join(sim.relpath, '*out*'))
+    if output_files:
+        with open(output_files[0]) as fh:
+            ns_day = parse_ns_day(fh)
+            fh.seek(0)
+            ncores = parse_ncores(fh)
+
+    # module = sim.categories['module']
+
+    return (sim.categories['module'], sim.categories['nodes'], ns_day,
+            sim.categories['time'], sim.categories['gpu'],
+            sim.categories['host'], ncores)
+
+
+def write_bench(top, tmpl, nodes, gpu, module, tpr, name, host, time):
+    """ Writes a single namd benchmark file and the respective sim object
+    """
+    sim = mds.Sim(
+        top['{}/'.format(nodes)],
+        categories={
+            'module': module,
+            'gpu': gpu,
+            'nodes': nodes,
+            'host': host,
+            'time': time,
+            'name': name,
+            'started': False
+        })
+
+    # Copy input files
+    namd = '{}.namd'.format(name)
+    psf = '{}.psf'.format(name)
+    pdb = '{}.pdb'.format(name)
+
+    with open(namd) as fh:
+        analyze_namd_file(fh)
+        fh.seek(0)
+
+    copyfile(namd, sim[namd].relpath)
+    copyfile(psf, sim[psf].relpath)
+    copyfile(pdb, sim[pdb].relpath)
+
+    # Add some time buffer to the requested time. Otherwise the queuing system
+    # kills the jobs before NAMD can finish
+    formatted_time = '{:02d}:{:02d}:00'.format(*divmod(time + 5, 60))
+    # engine switch to pick the right submission statement in the templates
+    md_engine = "namd"
+    # create bench job script
+    script = tmpl.render(
+        name=name,
+        gpu=gpu,
+        module=module,
+        mdengine=md_engine,
+        n_nodes=nodes,
+        time=time,
+        formatted_time=formatted_time)
+
+    with open(sim['bench.job'].relpath, 'w') as fh:
+        fh.write(script)
+
+
+def analyze_namd_file(fh):
+    """ Check whether the NAMD config file has any relative imports or variables
+    """
+    lines = fh.readlines()
+
+    for line in lines:
+        # Continue if we do not need to do anything with the current line
+        if ('parameters' not in line) and ('coordinates' not in line) and (
```

Comment:
@MSiggel I could not find your comment anymore. I just simplified your previous code, which repeated the `console.error` lines for all three cases.

If you are saying that we need to change the structure in general, then we should do this. Your previous code did not do it either.